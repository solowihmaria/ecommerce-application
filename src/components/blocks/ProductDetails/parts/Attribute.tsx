import styles from '../ProductDetails.module.scss';

export const Attribute = ({
    component,
    name,
    value,
}: {
    component: React.ReactNode;
    name: string;
    value: string | number;
}) => {
    return (
        <div className={styles.attributeContainer}>
            <span className={styles.iconContainer}>{component}</span>
            <div>
                <p className={styles.attributeText}>{name}</p>
                <p className={styles.attributeText}>{value}</p>
            </div>
        </div>
    );
};
