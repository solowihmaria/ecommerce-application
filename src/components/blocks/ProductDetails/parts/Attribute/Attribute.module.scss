@use '../../../../../assets/styles/constants.module' as c;

.attributeContainer {
    display: flex;
    flex-direction: row;
    gap: 0.875rem;
    align-items: center;
}

.iconContainer {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 40px;
    height: 40px;
    border: 1px solid white;
    border-radius: 50%;
}

.attributeText {
    margin: 0;
    color: c.$color-text-primary;
    text-transform: capitalize;
}

@media (width <= 992px) {
    .attributeContainer {
        min-width: 135px;
    }
}
