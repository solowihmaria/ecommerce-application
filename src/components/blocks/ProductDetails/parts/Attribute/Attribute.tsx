import styles from './Attribute.module.scss';
import type { AttributeProps } from './Attribute.types';

export const Attribute = ({ component, name, value }: AttributeProps) => {
    return (
        <div data-testid="attribute" className={styles.attributeContainer}>
            <span className={styles.iconContainer}>{component}</span>
            <div>
                <p className={styles.attributeText}>{name}</p>
                <p className={styles.attributeText}>{value}</p>
            </div>
        </div>
    );
};
