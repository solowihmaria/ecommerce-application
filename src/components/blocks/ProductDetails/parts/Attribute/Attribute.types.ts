export interface AttributeProps {
    component: React.ReactNode;
    name: string;
    value: string | number;
}
