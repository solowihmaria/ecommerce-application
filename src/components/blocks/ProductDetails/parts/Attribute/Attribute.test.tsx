import { render, screen } from '@testing-library/react';
import { Attribute } from './Attribute';
import '@testing-library/jest-dom';

const imageUrl = 'https://placehold.co/600x400/EEE/31343C';

describe('<Attribute/> component', () => {
    test('Render Attribute component', () => {
        render(
            <Attribute
                component={<img src={imageUrl} alt="height"></img>}
                name="Height"
                value={30}
            />
        );

        const attribute = screen.getByTestId('attribute');

        expect(attribute).toBeInTheDocument();
        expect(attribute).toMatchSnapshot();
    });
});
