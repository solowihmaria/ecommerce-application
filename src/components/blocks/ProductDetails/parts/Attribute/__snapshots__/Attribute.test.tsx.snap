// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SizeAttribute/> component Render Input component - without any properties 1`] = `
<div
  class="attributeContainer"
  data-testid="attribute"
>
  <span
    class="iconContainer"
  >
    <img
      alt="height"
      src="https://placehold.co/600x400/EEE/31343C"
    />
  </span>
  <div>
    <p
      class="attributeText"
    >
      Height
    </p>
    <p
      class="attributeText"
    >
      30
    </p>
  </div>
</div>
`;
